### PR TITLE
support CalcAmountIn interface

### PR DESCRIPTION
**Pull request by it4rb**

### CODE REVIEW

File: pkg/source/pool/iface.go [hunk 1]
```diff
@@ -50,6 +50,15 @@ type RFQResult struct {
 	Extra        any
 }
 
+type IPoolReverseSimulator interface {
```
Comment (hanguyenk) [inline pkg/source/pool/iface.go:53]: There are 2 trade types: exact in and exact out. I think we can name following the trade type: exactOutSimulator

File: pkg/source/pool/iface.go [hunk 2]
```diff
@@ -50,6 +50,15 @@ type RFQResult struct {
...
+	// CalcAmountIn calculate the `amountIn` of `tokenIn` needed to get `tokenAmountOut`
+	// caller might need to run `CalcAmountOut` again to determine if the returned `amountIn` is good enough
+	CalcAmountIn(
```
Comment (hanguyenk) [inline pkg/source/pool/iface.go:56]: Should we keep the interface similar to `CalcAmountOut`?

Comment (it4rb) [reply pkg/source/pool/iface.go:56]: um the result might be not be exact (the returned `amountIn` might not yield enough `amountOut`), so the caller might need to run CalcAmountOut again anyway to verify, I think they can get other param like `gas` from that 2nd call

File: pkg/source/pool/iface.go (hunk 1 above)
Comment (it4rb) [inline pkg/source/pool/iface.go:53]:
```suggestion
type IPoolExactOutSimulator interface {
```

**Review by hanguyenk — APPROVED** [review]

**Pull request merged**